openshift/e2e-tests-openshift.sh

```diff
@@ -184,7 +188,29 @@ spec:
   sourceNamespace: $OLM_NAMESPACE
   name: ${NAME}
   channel: techpreview
+  installPlanApproval: ${INSTALL_PLAN_APPROVAL}
+  startingCSV: ${UPGRADE_FROM_CSV}
 EOF
+
+  # Approve the initial installplan automatically
+  if [ $INSTALL_PLAN_APPROVAL = "Manual" ]; then
+    approve_csv $UPGRADE_FROM_CSV
+  fi
+}
+
+function approve_csv(){
+  local csv_version=$1
+
+  # Wait for the installplan to be available
+  timeout 300 "[[ \$(oc get InstallPlan -n $SERVING_NAMESPACE | grep -c $csv_version) -eq 0 ]]" || return 1
+
+  local install_plan=$(oc get InstallPlan -n $SERVING_NAMESPACE | grep $csv_version | awk '{ print $1}')
+  oc get InstallPlan $install_plan -n $SERVING_NAMESPACE -o yaml | sed 's/\(.*approved:\) false/\1 true/' | oc replace -f -
+  
+  # Wait for the installedCSV to be the new one
+  timeout 300 "[[ \$(oc get Subscription -n $SERVING_NAMESPACE -o jsonpath='{.items[0].status.installedCSV}' | grep -c $csv_version) -eq 0 ]]" || return 1
```

Comment:
Should we use single `'` here as well to be conformant? (Same in other places)

Reply:
I could try that but not sure how the variables would be resolved, e.g. $SERVING_NAMESPACE

Reply:
They are resolved at "runtime" of the condition (in each loop). You use the same below already for the `$serving_version`.

Reply:
Cool. I hate bash :-D